modules/es-cluster-health-is-red.adoc

```diff
@@ -0,0 +1,153 @@
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/settings?pretty
+----
+
+. If the `cluster.routing.allocation.enable` value is set to `none`, set it to `all`, by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/settings?pretty -X PUT -d '{"persistent": {"cluster.routing.allocation.enable":"all"}}'
+----
+
+. Check if any indices are still red by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+. If any indices are still red, try to clear them by performing the following steps.
+
+.. Clear the cache by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_cache/clear?pretty
+----
+
+.. Increase the max allocation retries by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.allocation.max_retries":10}'
+----
+
+.. Delete all the scroll items by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_search/scroll/_all -X DELETE
+----
+
+.. Increase the timeout by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.unassigned.node_left.delayed_timeout":"10m"}'
+----
+
+. If the preceding steps do not clear the red indices, delete the indices individually.
+
+.. Identify the red index name by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+.. Delete the red index by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_red_index_name> -X DELETE
+----
+
+. If there are no red indices and the cluster status is red, check for a continuous heavy processing load on a data node.
+
+.. Check if the Elasticsearch JVM Heap usage is high by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_nodes/stats?pretty
+----
++
+In the command output, review the `node_name.jvm.mem.heap_used_percent` field to determine the JVM Heap usage.
+
+.. Check for high CPU utilization.
```

Comment:
Maybe, is there more information that you want to provide?

Reply:
Maybe linking to [this page](https://docs.openshift.com/container-platform/4.13/monitoring/reviewing-monitoring-dashboards.html#reviewing-monitoring-dashboards-developer_reviewing-monitoring-dashboards) in the monitoring section is a good fit?

Reply:
Does this actually apply for ES? I.e. can you actually view CPU usage for it in the OpenShift console, or would it be in Kibana?

Reply:
Yes, from the POV of OpenShift, the Elasticsearch pods are a workload like any other, just that the access to the special `openshift-logging` namespace is usually restricted to a set of admins.

Reply:
We can't put a link here since it's a module and we'd need to xref to this other file, but I can mention here that they can check this in the web console.